inginious/frontend/plugins/task_hints/pages/user_hint_manager.py

```diff
@@ -0,0 +1,134 @@
+
+from inginious.frontend.parsable_text import ParsableText
+
+class UserHintManager(object):
+
+    """
+        Manage the user hints information in database, as the hints
+        that was unlocked, it's individual penalty, and cumulative
+        penalty, to be applied in the student's final grade for the
+        respective task.
+    """
+
+    def __init__(self, username, task_id, database):
+
+        self._username = username
+        self._task_id = task_id
+        self._database = database
+
+    def check_locked_hint_status(self, hints):
+
+        """ Method to check each hint status, and return the content
+            if it is allowed
```

Comment:
I'm not sure to understand what the word `allowed` means. Are they the locked or the unlocked ones? Please be concise with this, probably only using the words `locked` and `unlocked` is much understandable. 

Also, I am not sure to understand what this function is doing, the name is not good, does this just returns the unlocked hints? so a better name would be `get_unlocked_user_hints`? In case I am wrong, try to clarify this.